### ISSUE TITLE
Feature Request: Definable Healthpool Damage for NPCs

### ISSUE THREAD

**Issue by Saugia** [endless-sky/endless-sky#9269]
### Problem Description

Currently, we have an arbitrary way of denoting a "damaged ship" for a mission by removing some of its outfits or having a "broken" version of them (see: Remnant Bounty Missions). Additionally, it is also possible to denote a "damaged ship" by giving it a different total hull or shield value.

### Related Issue Links

I have not found any specific links pertaining to this feature request.

### Desired Solution

I'd like to have the ability to define how much of a ship's hull or shields is damaged. For example, a rescue mission could require you to escort a "damaged ship" from one location to another. Let's say that's a Behemoth, which has a pool of 7600 Shields and 6300 Hull. Instead of change the base hull values for the mission NPC, I'd like to be able to define something as such: 4900/6300 Hull. 

It might also be worth considering this if there are plans for future "support" weapons, such as giving shields to a friendly ship. This way, if a larger ship has slow shield recharge, or none at all, a player could "heal" the remaining value.

### Alternative Approaches

Alternatively, we could either add more types of "damaged outfits" like the Broke Jump Drive, or have some form of deprecation in terms of item usability, but that's a bit of a stretch.

### Additional Context

It is true that disabled ships is also used for this purpose, so this feature may not entirely be necessary, but I believe that having the flexibility between a full-health ship and a disabled ship for a mission or other NPC uses could benefit our content.

**Comment by DarcyManoel** [comment]
I don't see a strong use case for this feature and think a damaged ship is better just set to disabled.

**Comment by Saugia** [comment]
I disagree. Boarding a disabled ship in a given system and then escorting out is one playstyle, but it is not the same. With this feature, you could have escort missions where you meet up with your damage escort in a system, or your damaged escort jumps into a system you're in, and you have to escort it. These would not be possible with a disabled ship, and trailing off of that, I don't think it's very intuitive to go back and rely on changing base hull stats as that's more constraining a total hp pool, not implying damage to it.

**Comment by Hurleveur** [comment]
I always wanted to be able to define lower values than the max for escorts, beit for fuel, shields or hull - there's a use case in the unfettered campaign already where an Auxiliary that comes in should be a bit damaged but totally not disabled.
An npc with no fuel is also garanteed to land

**Comment by EjoThims** [comment]
I think it would make a lot more sense to do this as a multiplier applied to the standard shields/hull of the ship model.

Something like 'shield health .5' and 'hull health .3' (or whatever actual name we use) instead of requiring each use to write out the specifics of each as in the example given.

Probably more coding work, but much easier data implementation. The issue of accidentally giving too little hull and it being disabled automatically anyway is also easier to avoid, since you just have to ensure the multiplier you use is any larger than the disabled threshold instead of matching out what the actual hull value of that threshold is for whatever particular ship.

**Comment by Hurleveur** [comment]
Pushing this logic further, it could be useful to define a ship starting with 0 energy, or a ship that is ovetheated to begin with
With that in mind, I think a good way to approach this would be a new health section for ships, where every attribute can be changed based on a %

**Comment by Ferociousfeind** [comment]
For such modified ships/fleets, I would also want them to be frozen until the player enters the system they are in- otherwise you could camp outside a system with an overheated ship to allow it to cool off, or a badly-damaged ship that happens to have some hull repair to allow it to build up as much of a health buffer as possible.

And simply getting to the system the NPC is in might give them enough time to fix their problem themselves, without the player explicitly exploiting the "combat does not occur outside the system the player is in" thing.

**Comment by Hurleveur** [comment]
I'd just add an option for that
Being able to make npcs be frozen until you arrive in general would be good
